frontend/src/pages/calculatedResults/index.js

```diff
@@ -1,13 +1,91 @@
 import React from 'react';
-import { PlaceholderStyle } from '../../components/placeholderStyle/PlaceholderStyleElement';
-import './CalculatedResultsElements';
+import { useLocation } from 'react-router-dom';
+import Footer from '../../components/footerPurple';
+import {
+  FixedInstallmentsContainer,
+  IllustrationContainer,
+  DescendingInstallmentsContainer,
+  YearTabs,
+  Headers,
+} from './CalculatedResultsElements';
+import resultsIllustration from '../../images/results illustration.svg';
+import { SemiHeader } from '../../components/semiHeader/SemiHeader';
+import InstallmentContainer from '../../components/installmentContainer';
+import { PurpleAccentParagraph } from '../../components/purpleAccentParagraph/PurpleAccentParagraph';
+
+const { TabPane } = YearTabs;
 
 function CalculatedResults() {
+  const location = useLocation();
+
+  const getInterestType = (interestType) => {
+    if (interestType === 'fixed') return 'Oprocentowanie stałe';
+    return 'Oprocentowanie zmienne';
+  };
+
+  const fieldNames = [
+    'Rodzaj oprocentowania',
+    'Rodzaj rat',
+    'Wkład własny',
+    'Kwota kredytu',
+    'Długość trwania',
+    'Oprocentowanie',
+    'Prowizja',
+  ];
+
+  if (location.state.response?.user_input?.installment_type !== 'fixed') {
+    return (
+      <>
+        <DescendingInstallmentsContainer>
+          <YearTabs defaultActiveKey="1" centered>
+            <TabPane tab="Year 1" key="1">
+              Placeholder for descending installments
+            </TabPane>
+          </YearTabs>
+        </DescendingInstallmentsContainer>
+        <Footer />
+      </>
+    );
+  }
   return (
-    <PlaceholderStyle>
-      <h1>Placeholder for calculated results</h1>
-    </PlaceholderStyle>
+    <>
+      <FixedInstallmentsContainer>
+        <Headers>
+          <SemiHeader>Twoja rata kredytu</SemiHeader>
+          <PurpleAccentParagraph>
+            {getInterestType(location.state.response.user_input.interest_type)}
+          </PurpleAccentParagraph>
+        </Headers>
+        <InstallmentContainer
```

Comment:
Could be done with model + `map()`

Reply:
https://www.joshualyman.com/2018/01/simple-javascript-model-mapper-utility-functions/ the source I find on model + mapping is this one and I am not sure if this is what you mean. For time being I solved it with a ternary operator

Reply:
I mean something that you did before ;) Just create a model array with the values and map them to the JSX to avoid code repeating and compact it.

```
const installmentModels = [
  {
    title: 'Twoje koszty kredytu',
    type: 'baseline_time',
    data: location.state.response.baseline_time,
    addon: 'zł'
  },
  {
    ...
  }
];
```

And somewhere in JSX:

```
installmentModels.map(model => (<InstallmentContainer {...model} />)
```